### PR TITLE
Plugin centric marketplace integration

### PR DESCRIPTION
**Pull request by ddc22**
#### Changes proposed in this Pull Request

- This PR brings together several PRs which split the bigger PR  #54290, the previous PRs are
  - #54489
  - #54552
  - #54585

- Some new tests for marketplace/selectors were also added

#### Testing instructions
* Make sure, Flow 1,2 in decision grid, before and after atomic should work.

Related to #54290

### CODE REVIEW

File: client/my-sites/marketplace/pages/marketplace-product-details/index.tsx [hunk 1]
```diff
@@ -49,39 +45,41 @@ interface MarketplacePluginDetailsInterface {
 
 function MarketplacePluginDetails( {
 	productGroupSlug,
+	productSlug,
 }: MarketplacePluginDetailsInterface ): JSX.Element {
-	// TODO : Depend on product slug to show product details in the page
-	const productSlug = 'yoast_premium';
-	const marketplacePluginSlug = getDefaultPluginInProduct( productGroupSlug, productSlug );
+	/* ****** TODO : Depend on product slug to show product details in this page ****** */
+	const hardcodedProductSlug = 'yoast_premium';
```
Comment (ddc22) [inline client/my-sites/marketplace/pages/marketplace-product-details/index.tsx:52]: This will be changed in the next PRs where Yoast will be removed.

**Review by dzver — APPROVED** [review]
LGTM, thanks for working on this!

File: client/my-sites/marketplace/pages/marketplace-plugin-setup-status/index.tsx [hunk 2]
```diff
@@ -76,23 +78,36 @@ function WrappedMarketplacePluginSetup(): JSX.Element {
...
-				'::MARKETPLACE::ERROR:: There is an error in plugin setup page pluginSlugToBeInstalled is not provided'
+				'::MARKETPLACE::ERROR:: There is an error in plugin setup page, productSlugInstalled or productGroupSlug is not provided'
 			);
 			page( `/home/${ selectedSiteSlug }` );
+		} else {
+			const plugins = getPluginsToInstall( productGroupSlug, productSlugInstalled );
+			if ( ! Array.isArray( plugins ) || plugins.length === 0 ) {
+				marketplaceDebugger(
+					`::MARKETPLACE::ERROR:: There is an error in plugin setup page, plugins to install are not available ${
+						( productGroupSlug, productSlugInstalled )
```
Comment (tyxla) [inline client/my-sites/marketplace/pages/marketplace-plugin-setup-status/index.tsx:101]: Noting that `( productGroupSlug, productSlugInstalled )` will evaluate to just the value of `productSlugInstalled`. I'd bet you didn't mean this, but rather `[ 'productGroupSlug', 'productSlugInstalled' ]` or something similar that would just concatenate the two values?

Comment (ddc22) [reply client/my-sites/marketplace/pages/marketplace-plugin-setup-status/index.tsx:101]: Ah yes this is incorrect, this should be replaced with below, nice catch 👍 

`::MARKETPLACE::ERROR:: There is an error in plugin setup page, plugins to install are not available 
${productGroupSlug}, ${productSlugInstalled}`

Are you working on this part of the code? If not I can whip up a PR :)

Comment (tyxla) [reply client/my-sites/marketplace/pages/marketplace-plugin-setup-status/index.tsx:101]: Not working on it right now, I was just looking at code that's been merged while I was absent 😉 

Thanks, I'd be happy to help review that PR 😉

**Pull request merged**